compiler/src/dmd/errors.d

```diff
@@ -122,33 +122,67 @@ void generateSarifReport(const ref Loc loc, const(char)* format, va_list ap, Err
     // Create an OutBuffer to store the SARIF report
     OutBuffer ob;
 
+    // Extract and clean the version string
+    const(char)* rawVersionChars = global.versionChars();
+
+    // Remove 'v' prefix if it exists
+    if (*rawVersionChars == 'v') {
+        rawVersionChars += 1;
+    }
+
+    // Find the first non-numeric character after the version number
+    const(char)* nonNumeric = strchr(rawVersionChars, '-');
+    size_t length;
+    if (nonNumeric !is null) {
+        length = cast(size_t)(nonNumeric - rawVersionChars);
+    } else {
+        length = strlen(rawVersionChars);
+    }
+
     // Build SARIF report using OutBuffer
     ob.writestring("{\n");
-    ob.writestring("  \"invocation\": {\n");
-    ob.writestring("    \"executionSuccessful\": false\n");
-    ob.writestring("  },\n");
-    ob.writestring("  \"results\": [\n    {\n");
-
-    // Write location information with relative path
-    ob.writestring("      \"location\": {\n");
-    ob.writestring("        \"artifactLocation\": {\n");
-    ob.printf("          \"uri\": \"%s\"\n", loc.filename);
-    ob.writestring("        },\n");
-    ob.writestring("        \"region\": {\n");
-    ob.printf("          \"startLine\": %d,\n", loc.linnum);
-    ob.printf("          \"startColumn\": %d\n", loc.charnum);
+    ob.writestring("  \"version\": \"2.1.0\",\n");
```

Comment:
```suggestion
    ob.writestringln(`  "version": "2.1.0",`);
```
There's little need to use escape sequences everywhere.

Reply:
fixed- https://github.com/dlang/dmd/pull/17020/commits/7a742fe086f92c4266ad6f2e0740d044fafe24d5